test/stress-test.js

```diff
@@ -0,0 +1,743 @@
+      throw Error(
+        `Expect context.readServersWithQueryCount[address] to be greater then 1, but it is ${context.readServersWithQueryCount[address]}`
+      )
+    }
+    if (context.writeServersWithQueryCount[address] <= 1) {
+      throw Error(
+        `Expect context.writeServersWithQueryCount[address] to be greater then 1, but it is ${context.writeServersWithQueryCount[address]}`
+      )
+    }
+
+    resolve()
+  })
+}
+
+function verifyCausalClusterMembers (context) {
+  return fetchClusterAddresses(context).then(clusterAddresses => {
+    // Before 3.2.0 only read replicas serve reads.
+    // We round up this check to Bolt v2 and above
+    // to avoid sniffing the server agent string.
+    const readsOnFollowersEnabled = context.protocolVersion >= 2
+
+    if (readsOnFollowersEnabled) {
+      // expect all followers to serve more than zero read queries
+      assertAllAddressesServedReadQueries(
+        clusterAddresses.followers,
+        context.readServersWithQueryCount
+      )
+    }
+
+    // expect all read replicas to serve more than zero read queries
+    assertAllAddressesServedReadQueries(
+      clusterAddresses.readReplicas,
+      context.readServersWithQueryCount
+    )
+
+    if (readsOnFollowersEnabled) {
+      // expect all followers to serve same order of magnitude read queries
+      assertAllAddressesServedSimilarAmountOfReadQueries(
+        clusterAddresses.followers,
+        context.readServersWithQueryCount
+      )
+    }
+
+    // expect all read replicas to serve same order of magnitude read queries
+    assertAllAddressesServedSimilarAmountOfReadQueries(
+      clusterAddresses.readReplicas,
+      context.readServersWithQueryCount
+    )
+  })
+}
+
+function fetchClusterAddresses (context) {
+  const session = context.driver.session()
+  return session
+    .readTransaction(tx => tx.run('CALL dbms.cluster.overview()'))
+    .then(result => {
+      const records = result.records
+      const supportsMultiDb = context.protocolVersion >= 4.0
+      const followers = supportsMultiDb
+        ? addressesForMultiDb(records, 'FOLLOWER')
+        : addressesWithRole(records, 'FOLLOWER')
+      const readReplicas = supportsMultiDb
+        ? addressesForMultiDb(records, 'READ_REPLICA')
+        : addressesWithRole(records, 'READ_REPLICA')
+
+      return session
+        .close()
+        .then(() => new ClusterAddresses(followers, readReplicas))
+    })
+}
+
+function addressesForMultiDb (records, role, db = 'neo4j') {
+  return _.uniq(
+    records
+      .filter(record => record.get('databases')[db] === role)
+      .map(record => record.get('addresses')[0].replace('bolt://', ''))
+  )
+}
+
+function addressesWithRole (records, role) {
+  return _.uniq(
+    records
+      .filter(record => record.get('role') === role)
+      .map(record => record.get('addresses')[0].replace('bolt://', ''))
+  )
+}
+
+function assertAllAddressesServedReadQueries (addresses, readQueriesByServer) {
```

Comment:
I'm don't believe that is safe to do this check if the cluster is reforming while the tests are running we might end up with new members of the cluster right before this check.
Load balancing algorithm is better tested with other means..

Reply:
Sure, I will remove all the `verifyCausalClusterMembers` function, since it's could cause problem.